src/core_editor/line_buffer.rs

```diff
@@ -179,6 +179,24 @@ impl LineBuffer {
             .unwrap_or_else(|| self.lines.len())
     }
 
+    /// Cursor position *at end of* the next word to the right
+    pub fn word_right_end_index(&self) -> usize {
+        self.lines[self.insertion_point..]
+            .split_word_bound_indices()
+            .find(|(i, word)| *i + word.len() > 1 && !is_whitespace_str(word))
+            .map(|(i, word)| self.insertion_point + i + word.len() - 1)
+            .unwrap_or_else(|| self.lines.len().saturating_sub(1))
```

Comment:
Keep in mind that UTF-8 encoded strings can contain characters longer than 1 byte, and further string operations would panic if the character boundaries are violated.

Test case:

`word😇 with emoji`
go to insertion point `0`, hit `e`

We currently operate on the grapheme level i.e. multiple code characters that get displayed as one character on the screen (e.g. accents on some European scripts, joined emojis (most terminals actually don't fuse them), CRLF etc.)

```suggestion
        self.lines[..self.word_right_index()]
            .grapheme_indices(true)
            .next_back()
            .map_or(0, |(idx, _)| idx)
```

Reply:
Sorry the particular suggestion will get stuck on the last character.

Reply:
new version should fix it